modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java

```diff
@@ -621,6 +621,12 @@ private Schema modelFromProperty(OpenAPI openAPI, Schema object, String path) {
         XML xml = object.getXml();
         Map<String, Schema> properties = object.getProperties();
         Schema model = new Schema();
+        if (object.getType() != null) {
+            model.setType(object.getType());
+        }
+        if (object.getFormat() != null) {
+            model.setFormat(object.getFormat());
```

Comment:
IMO, there may be a contradiction in the OAS spec (both 3.0.3 and 3.1) versus the JSON schema specification.
I have opened https://github.com/OAI/OpenAPI-Specification/issues/2301. It would be interesting to have @wing328  and @jimschubert  opinions.
In any case, I suggest adding the code comments below because later it can be hard for maintainers to understand the context and why this was done.
```suggestion
            // Even though the `format` keyword typically applies to primitive types only,
            // the JSON schema specification states `format` can be used for any model type instance.
            model.setFormat(object.getFormat());
```